src/test/scala/org/apache/spark/sql/delta/MergeIntoScalaSuite.scala

```diff
@@ -279,6 +280,88 @@ class MergeIntoScalaSuite extends MergeIntoSuiteBase {
     }
   }
 
+  test("findTouchedFiles - no files match target predicate") {
```

Comment:
Any reason why this is in MergeIntoScalaSuite and not in the MergeIntoSuiteBase? 
The reason that these exists is that MergeIntoSuiteBase contains tests that should run on merge executed on both, SQL and Scala APIs. We internally have MergeIntoScalaSuite and MergeIntoSQLSuite that extend MergeIntoSuiteBase. We will expose MergeIntoSQLSuite as soon as we migrate oss delta depend on spark 3.0 which will bring in merge SQL support in the spark parser. So, the bottom line is, if there are features that are independent of the language invoking the merge, please put their tests in MergeIntoSuiteBase so that both APIs get tested for that feature.

Reply:
As long as there isn't value in testing differences that can be caused by API differences (different analysis code path, different resolution code path), I don't see the value in doubling the unit test time. I feel this tests a specific condition of the merge, independent of API or resolution, therefore it's fine to keep it in a single test suite to not 2x test time. What do you think?

Reply:
While I agree that, in general, it is unnecessary to test this twice. However, given the complexity of tests related to this most complex operation, merge, it's better to enforce a cleaner and easier-to-reason-about tests. The moment we start separating these tests by such subtle logic, we will start adding some tests in one file, other in the other files, thus opening up to future questions like 
1. Where is the test that is supposed to test this? in SuiteBase, SQLSuite, ScalaSuite? 
2. X was tested in ScalaSuite. Are you sure that internally SQL will follow the exact same code path in the future?

A bit more testing time is worth it for achieving a much easier-to-reason-and-maintain testing matrix.

Reply:
@brkyvz @tdas I wrestled with putting in SuiteBase vs. ScalaSuite, but I wasn't able to clearly define what belongs in the SuiteBase.

Reply:
I think now you know :)